src/Dhall/Import.hs

```diff
@@ -508,128 +508,154 @@ instance Show HashMismatch where
         <>  "\n"
         <>  "↳ " <> show actualHash <> "\n"
 
--- | Parse an expression from a `Import` containing a Dhall program
-exprFromImport :: Import -> StateT Status IO (Expr Src Import)
-exprFromImport (Import {..}) = do
-    let ImportHashed {..} = importHashed
-
-    (path, text) <- case importType of
-        Local prefix (File {..}) -> liftIO $ do
-            let Directory {..} = directory
-
-            prefixPath <- case prefix of
-                Home -> do
-                    System.Directory.getHomeDirectory
-
-                Absolute -> do
-                    return "/"
+-- | A path made absolute
+data ResolvedPath
+    = LocalPath FilePath
+    | UrlPath Text (Maybe ImportHashed)
+    | EnvPath Text
 
-                Here -> do
-                    System.Directory.getCurrentDirectory
-
-            let cs = map Text.unpack (file : components)
-
-            let cons component dir = dir </> component
-
-            let path = foldr cons prefixPath cs
-
-            exists <- System.Directory.doesFileExist path
-
-            if exists
-                then return ()
-                else throwIO (MissingFile path)
+instance Show ResolvedPath where
+    show (LocalPath file) = file
+    show (UrlPath url _) = Text.unpack url
+    show (EnvPath env) = Text.unpack env
 
-            text <- Data.Text.IO.readFile path
+-- | Exception thrown when @importType@ and @importMode@ are incompatible
+data IllegalImportType = EnvAsLocation deriving (Typeable)
 
-            return (path, text)
+instance Exception IllegalImportType
 
-        URL prefix file suffix maybeHeaders -> do
-            m <- needManager
+instance Show IllegalImportType where
+    show EnvAsLocation = _ERROR <> ": env cannot be imported as a location"
```

Comment:
`s/env/an environment variable/` and `s/as location/❰as Location❱/`

Reply:
✅